Cannot find -lfftw3 with SPEC_ROOT on Ocean

# Bug reports:

Expecting -lfftw3 to be found when `SPEC_ROOT` is set.
Currently linking only succeeds with `SPEC_ROOT` is unset.
@nilsvu says this is related to @nilsdeppe 's #5250 

This is on Ocean with gcc.

<!--
describe the expected behavior
-->

### Current behavior:
When building,
`/usr/bin/ld.gold: error: cannot find -lfftw3`

### Environment:

GSL version: 2.7.1
HDF5 version: 1.12.2
jemalloc version: 5.2.1
SpEC exporter: /opt/ohpc/pub/apps/spec/spec-interp4/Support/ApplyObservers/Exporter
Catch version: 2.13.8
NumPy version: 1.21.1
CMake version: 3.24.1
CMAKE_CXX_FLAGS:
CMAKE_CXX_LINK_FLAGS:  -fuse-ld=gold
CMAKE_CXX_FLAGS_DEBUG: -g -DSPECTRE_DEBUG
CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS:
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_Fortran_FLAGS:
CMAKE_Fortran_FLAGS_DEBUG: -g
CMAKE_Fortran_FLAGS_RELEASE: -O3 -DNDEBUG -O3
CMAKE_EXE_LINKER_FLAGS:
CMAKE_BUILD_TYPE: Release
CMAKE_CXX_COMPILER: /opt/ohpc/pub/compiler/gcc/11.3.0/bin/g++
CMAKE_CXX_COMPILER_VERSION: 11.3.0
CMAKE_C_COMPILER: /opt/ohpc/pub/compiler/gcc/11.3.0/bin/gcc
CMAKE_C_COMPILER_VERSION: 11.3.0
CMAKE_Fortran_COMPILER: /opt/ohpc/pub/compiler/gcc/11.3.0/bin/gfortran
CMAKE_Fortran_COMPILER_VERSION: 11.3.0
Python version: 3.9.5

--- a/cmake/FindFFTW.cmake
+++ b/cmake/FindFFTW.cmake
@@ -1,0 +1,36 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Find FFTW3 (The Fastest Fourier Transform in the West)
+# http://www.fftw.org/
+#
+# Defines the `FFTW::FFTW` target to link against.
+
+if(NOT FFTW_ROOT)
+  # Need to set to empty to avoid warnings with --warn-uninitialized
+  set(FFTW_ROOT "")
+  set(FFTW_ROOT $ENV{FFTW_ROOT})
+endif()
+
+find_path(FFTW_INCLUDE_DIR
+  NAMES fftw3.h
+  PATH_SUFFIXES include
+  HINTS ${FFTW_ROOT}
+  )
+
+find_library(FFTW_LIB
+  NAMES fftw3
+  PATH_SUFFIXES lib64 lib
+  HINTS ${FFTW_ROOT}
+  )
+
+if (FFTW_INCLUDE_DIR AND FFTW_LIB)
+  add_library(FFTW::FFTW INTERFACE IMPORTED)
+  target_include_directories(FFTW::FFTW INTERFACE ${FFTW_INCLUDE_DIR})
+  target_link_libraries(FFTW::FFTW INTERFACE ${FFTW_LIB})
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(FFTW
+  REQUIRED_VARS FFTW_INCLUDE_DIR FFTW_LIB)
+mark_as_advanced(FFTW_INCLUDE_DIR FFTW_LIB)

--- a/cmake/FindSpEC.cmake
+++ b/cmake/FindSpEC.cmake
@@ -45,33 +45,6 @@ find_package(MPI COMPONENTS C)
 
 if (SPEC_PACKAGED_EXPORTER_LIB AND SPEC_EXPORTER_FACTORY_OBJECTS AND
     SPEC_EXPORTER_INCLUDE_DIR AND MPI_C_FOUND)
-
-  # Deal with FFTW3
-  #
-  # If it was dynamically linked into SpEC then we need to dynamically link it
-  # into SpECTRE.
-  set(_MACHINE_DEF_FILE
-    ${SPEC_ROOT}/MakefileRules/this_machine.def)
-  set(_FFTW3_LINK_FLAGS "")
-  if (EXISTS ${_MACHINE_DEF_FILE})
-    file(STRINGS ${_MACHINE_DEF_FILE} _MACHINE_DEF)
-    # Filter starting comments
-    list(FILTER _MACHINE_DEF EXCLUDE REGEX "^[ ]*#")
-    # Remove inline comments
-    list(TRANSFORM _MACHINE_DEF REPLACE "[ ]*#.*" "")
-    string(REGEX MATCH "FFTW_LIB[^;]+"
-      _FFTW3_DEF "${_MACHINE_DEF}")
-    string(REGEX REPLACE
-      "FFTW_LIB[ +]+=[ ]" ""
-      _FFTW3_LINK_FLAGS
-      ${_FFTW3_DEF})
-    string(FIND ${_FFTW3_DEF} "-lfftw3" _FOUND_FFTW3_SHARED)
-    # If we did not find a shared lib, do not add linking flags.
-    if(${_FOUND_FFTW3_SHARED} STREQUAL -1)
-      set(_FFTW3_LINK_FLAGS "")
-    endif()
-  endif()
-
   add_library(SpEC::Exporter INTERFACE IMPORTED)
   target_include_directories(
     SpEC::Exporter INTERFACE ${SPEC_EXPORTER_INCLUDE_DIR})
@@ -87,8 +60,29 @@ if (SPEC_PACKAGED_EXPORTER_LIB AND SPEC_EXPORTER_FACTORY_OBJECTS AND
     # The order of these next two lines is important
     ${SPEC_EXPORTER_FACTORY_OBJECTS}
     ${SPEC_PACKAGED_EXPORTER_LIB}
-    ${_FFTW3_LINK_FLAGS}
   )
+
+  # Deal with FFTW3
+  #
+  # If it was dynamically linked into SpEC then we need to dynamically link it
+  # into SpECTRE.
+  set(_MACHINE_DEF_FILE
+    ${SPEC_ROOT}/MakefileRules/this_machine.def)
+  if (EXISTS ${_MACHINE_DEF_FILE})
+    file(STRINGS ${_MACHINE_DEF_FILE} _MACHINE_DEF)
+    # Filter starting comments
+    list(FILTER _MACHINE_DEF EXCLUDE REGEX "^[ ]*#")
+    # Remove inline comments
+    list(TRANSFORM _MACHINE_DEF REPLACE "[ ]*#.*" "")
+    string(REGEX MATCH "FFTW_LIB[^;]+"
+      _FFTW3_DEF "${_MACHINE_DEF}")
+    string(FIND ${_FFTW3_DEF} "-lfftw3" _FOUND_FFTW3_SHARED)
+    # If we found a shared FFTW lib in SpEC, find and link FFTW here
+    if(NOT ${_FOUND_FFTW3_SHARED} STREQUAL "-1")
+      find_package(FFTW REQUIRED)
+      target_link_libraries(SpEC::Exporter INTERFACE FFTW::FFTW)
+    endif()
+  endif()
 endif()
 
 include(FindPackageHandleStandardArgs)

